sdk/storage/Azure.Storage.DataMovement/src/Shared/JobPlanExtensions.cs

```diff
@@ -120,7 +101,37 @@ internal static JobPartPlanHeader GetJobPartPlanHeader(this JobPartPlanFileName
                     }
                 }
             }
-            return storedPath.TrimEnd('\\', '/');
+
+            if (partCount == 1)
+            {
+                return storedPath;
+            }
+            else
+            {
+                int lastSlash = storedPath.Replace('\\', '/').LastIndexOf('/');
+                return storedPath.Substring(0, lastSlash);
+            }
```

Comment:
So yeah, this was a strange one. Firstly, the existing was not actually being used so we're not breaking anything existing. This code is intendeds to find the longest common path between all the job parts (since we don't have a main file that keeps track of top-level transfer info yet) but the existing code would find the longest common **string**.

For example:
`C:\Test\file1`
`C:\Test\file2`

This would result in `C:\Test\file` when we really want just `C:\Test`.

So, my thought was to just trim back to the last slash BUT we only want to do this if there are multiple job parts (multiple files). If there is only one, the path we need to return is to a specific resource but if there are more than one, we will never return a path to a specific resource since we are guaranteed there will be a parent directory. I think this makes it safe to always trim back to the last slash in the latter case.

There are obviously more sophisticated ways to do this that may be needed in more complex scenarios, but **I think** this will work and is rather simple and fast. Python offers [commonpath](https://docs.python.org/3/library/os.path.html#os.path.commonpath) but I couldn't find something in .NET.